wire/encode.go

```diff
@@ -117,7 +117,7 @@ func (mb *MessageBuilder) WriteArray(v []byte) {
 	}
 }
 
-func (mb *MessageBuilder) WriteFile(v *os.File) {
+func (mb *MessageBuilder) WriteFile(v FileDescriptor) {
 	fd, err := unix.Dup(int(v.Fd()))
```

Comment:
No, I mean doing this:

```suggestion
func (mb *MessageBuilder) WriteFile(file *os.File) {
	mb.WriteFD(int(file.Fd()))
}

func (mb *MessageBuilder) WriteFD(file int) {
	fd, err := unix.Dup(file)
```